classroom/snippets/src/main/java/CancelGuardianInvitation.java

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START classroom_cancel_guardian_invitation_class]
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.googleapis.json.GoogleJsonError;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.services.classroom.Classroom;
+import com.google.api.services.classroom.ClassroomScopes;
+import com.google.api.services.classroom.model.GuardianInvitation;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+/* Class to demonstrate the use of Classroom Patch Guardian Invitation API. */
+public class CancelGuardianInvitation {
+  /**
+   * Cancel a guardian invitation by modifying the state of the invite.
+   *
+   * @param studentId - the id of the student.
+   * @param invitationId - the id of the guardian invitation to modify.
+   * @return - the modified guardian invitation.
+   * @throws IOException - if credentials file not found.
+   */
+  public static GuardianInvitation cancelGuardianInvitation(String studentId, String invitationId)
+      throws Exception {
+
+    /* Scopes required by this API call. If modifying these scopes, delete your previously saved
+    tokens/ folder. */
+    final List<String> SCOPES =
+        Collections.singletonList(ClassroomScopes.CLASSROOM_GUARDIANLINKS_STUDENTS);
+
+    // Create the classroom API client
+    ClassroomCredentials classroomCredentials = new ClassroomCredentials();
```

Comment:
Yeah so the application default credentials don't work when requesting sensitive scopes and I haven't been able to write tests that work for guardians/guardian invites because of this. So I'm going to be updating all of the classes to use this type of credentialing in Q1. I've just started out with the Guardian classes because there is no way to test them without this.